scylladb don't compile on nixos

```
[larluo@nixos-larluo:~]$ nix-env -f '<nixpkgs-unstable>' -iA scylladb
installing 'scylladb-3.0.5'
these derivations will be built:
  /nix/store/lp8phrjp1g1fmmh1cfp8jrz7q3a7f505-scylladb-3.0.5.drv
building '/nix/store/lp8phrjp1g1fmmh1cfp8jrz7q3a7f505-scylladb-3.0.5.drv'...
unpacking sources
unpacking source archive /nix/store/2nr9978qp56qnmrl4phm1d8p9l4dhkp3-scylla-403f66e
source root is scylla-403f66e
patching sources
applying patch /nix/store/4s4hblyppc69qjk8133ahx9p4xsf9s0z-seastar-configure-script-paths.patch
patching file seastar/configure.py
patching script interpreter paths in ./configure.py
./configure.py: interpreter directive changed from "/usr/bin/python3" to "/nix/store/swy0p01xr0wyh907d67hkxr1g0kngcpn-python3-3.7.4/bin/python3"
configuring
Traceback (most recent call last):
  File "./configure.py", line 38, in <module>
    for line in open('/etc/os-release'):
FileNotFoundError: [Errno 2] No such file or directory: '/etc/os-release'
builder for '/nix/store/lp8phrjp1g1fmmh1cfp8jrz7q3a7f505-scylladb-3.0.5.drv' failed with exit code 1
error: build of '/nix/store/lp8phrjp1g1fmmh1cfp8jrz7q3a7f505-scylladb-3.0.5.drv' failed

[larluo@nixos-larluo:~]$ cat /etc/os-release
NAME=NixOS
ID=nixos
VERSION="19.09pre177248.ed1b59a98e7 (Loris)"
VERSION_CODENAME=loris
VERSION_ID="19.09pre177248.ed1b59a98e7"
PRETTY_NAME="NixOS 19.09pre177248.ed1b59a98e7 (Loris)"
LOGO="nix-snowflake"
HOME_URL="https://nixos.org/"
SUPPORT_URL="https://nixos.org/nixos/support.html"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
```

`scylladb` seems to be broken in more than one way. I get (also on `nixos-unstable`)

```
...
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j4 -l4
[1/577] SWAGGER build/release/gen/api/api-doc/storage_service.json.hh
FAILED: build/release/gen/api/api-doc/storage_service.json.hh
seastar/json/json2code.py -f api/api-doc/storage_service.json -o build/release/gen/api/api-doc/storage_service.json.hh
/bin/sh: seastar/json/json2code.py: not found
[2/577] SWAGGER build/release/gen/api/api-doc/lsa.json.hh
FAILED: build/release/gen/api/api-doc/lsa.json.hh
seastar/json/json2code.py -f api/api-doc/lsa.json -o build/release/gen/api/api-doc/lsa.json.hh
/bin/sh: seastar/json/json2code.py: not found
[3/577] SWAGGER build/release/gen/api/api-doc/commitlog.json.hh
FAILED: build/release/gen/api/api-doc/commitlog.json.hh
seastar/json/json2code.py -f api/api-doc/commitlog.json -o build/release/gen/api/api-doc/commitlog.json.hh
/bin/sh: seastar/json/json2code.py: not found
[4/577] SWAGGER build/release/gen/api/api-doc/gossiper.json.hh
FAILED: build/release/gen/api/api-doc/gossiper.json.hh
seastar/json/json2code.py -f api/api-doc/gossiper.json -o build/release/gen/api/api-doc/gossiper.json.hh
/bin/sh: seastar/json/json2code.py: not found
ninja: build stopped: subcommand failed.
```

Pinging maintainer: @workflow

> `scylladb` seems to be broken in more than one way. I get (also on `nixos-unstable`)
> 
> ```
> ...
> @nix { "action": "setPhase", "phase": "buildPhase" }
> building
> build flags: -j4 -l4
> [1/577] SWAGGER build/release/gen/api/api-doc/storage_service.json.hh
> FAILED: build/release/gen/api/api-doc/storage_service.json.hh
> seastar/json/json2code.py -f api/api-doc/storage_service.json -o build/release/gen/api/api-doc/storage_service.json.hh
> /bin/sh: seastar/json/json2code.py: not found
> [2/577] SWAGGER build/release/gen/api/api-doc/lsa.json.hh
> FAILED: build/release/gen/api/api-doc/lsa.json.hh
> seastar/json/json2code.py -f api/api-doc/lsa.json -o build/release/gen/api/api-doc/lsa.json.hh
> /bin/sh: seastar/json/json2code.py: not found
> [3/577] SWAGGER build/release/gen/api/api-doc/commitlog.json.hh
> FAILED: build/release/gen/api/api-doc/commitlog.json.hh
> seastar/json/json2code.py -f api/api-doc/commitlog.json -o build/release/gen/api/api-doc/commitlog.json.hh
> /bin/sh: seastar/json/json2code.py: not found
> [4/577] SWAGGER build/release/gen/api/api-doc/gossiper.json.hh
> FAILED: build/release/gen/api/api-doc/gossiper.json.hh
> seastar/json/json2code.py -f api/api-doc/gossiper.json -o build/release/gen/api/api-doc/gossiper.json.hh
> /bin/sh: seastar/json/json2code.py: not found
> ninja: build stopped: subcommand failed.
> ```
> 
> Pinging maintainer: @workflow

Not sure how to replicate this one yet

--- a/pkgs/servers/scylladb/configure-etc-osrelease.patch
+++ b/pkgs/servers/scylladb/configure-etc-osrelease.patch
@@ -1,0 +1,28 @@
+diff --git a/configure.py b/configure.py
+index 25ca951ac..454140420 100755
+--- a/configure.py
++++ b/configure.py
+@@ -36,13 +36,16 @@ tempfile.tempdir = "./build/tmp"
+
+ configure_args = str.join(' ', [shlex.quote(x) for x in sys.argv[1:]])
+
+-for line in open('/etc/os-release'):
+-    key, _, value = line.partition('=')
+-    value = value.strip().strip('"')
+-    if key == 'ID':
+-        os_ids = [value]
+-    if key == 'ID_LIKE':
+-        os_ids += value.split(' ')
++try:
++    for line in open('/etc/os-release'):
++        key, _, value = line.partition('=')
++        value = value.strip().strip('"')
++        if key == 'ID':
++            os_ids = [value]
++        if key == 'ID_LIKE':
++            os_ids += value.split(' ')
++except FileNotFoundError:
++    os_ids = ["linux"]
+
+
+ # distribution "internationalization", converting package names.

--- a/pkgs/servers/scylladb/default.nix
+++ b/pkgs/servers/scylladb/default.nix
@@ -83,10 +83,12 @@ gcc8Stdenv.mkDerivation {
   configurePhase = ''
     ./configure.py --mode=release
   '';
+
   installPhase = ''
     mkdir $out
     cp -r * $out/
   '';
+
   meta = with stdenv.lib; {
     description = "NoSQL data store using the seastar framework, compatible with Apache Cassandra";
     homepage = "https://scylladb.com";

--- a/pkgs/servers/scylladb/default.nix
+++ b/pkgs/servers/scylladb/default.nix
@@ -77,6 +77,7 @@ gcc8Stdenv.mkDerivation {
 
   postPatch = ''
     patchShebangs ./configure.py
+    patchShebangs seastar/json/json2code.py
   '';
 
   configurePhase = ''

--- a/pkgs/servers/scylladb/default.nix
+++ b/pkgs/servers/scylladb/default.nix
@@ -41,7 +41,7 @@ gcc8Stdenv.mkDerivation {
     fetchSubmodules = true;
   };
 
-  patches = [ ./seastar-configure-script-paths.patch ];
+  patches = [ ./seastar-configure-script-paths.patch ./configure-etc-osrelease.patch ];
 
   nativeBuildInputs = [
    pkgconfig

